pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -794,7 +901,22 @@ func (p *PriorityQueue) movePodsToActiveOrBackoffQueue(podInfoList []*framework.
 			}
 		}
 	}
-	p.moveRequestCycle = p.schedulingCycle
+
+	// Besides setting MoveRequestCycle above for all pods which got moved,
+	// we also need to set it for pods that are being processed because
+	// AddUnschedulableIfNotPresent might get called for them, and in
+	// AddUnschedulableIfNotPresent we need to know whether events where
+	// observed while processing them.
+	for uid, inFlight := range p.inFlightPods {
+		if !p.runRequeuePlugins(ctx, inFlight.pod, &event, oldObj, newObj) {
+			// This Pod isn't worth putting to activeQ or backoffQ.
```

Comment:
I have implemented `dynamicResources.Requeue` and ran the integration test. I see cases where `inFlight.moveRequestCycle` gets set because of an updated `ResourceClaim` without first calling  `dynamicResources.Requeue`, probably because that plugin wasn't in `UnschedulablePlugins` yet. The effect is that the pod got moved to the backoff queue although the concurrent cluster event probably wasn't relevant for it.

I just don't see how this call to `runRequeuePlugins` can avoid false positives for concurrent cluster events.